BookingProject/BookingWebAPI/AuditWebApi.Application/AuditService.cs

```diff
@@ -1,30 +1,35 @@
 ﻿using AuditWebApi.Application.DTOs;
 using AuditWebApi.Domain.Constants;
 using AuditWebApi.Domain.Entities;
-using AutoMapper;
+using Contracts.DTOs;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Response;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Text.Json;
-using System.Threading.Tasks;
+using System.Net.Http.Json;
+
 
 namespace AuditWebApi.Application
 {
     public interface IAuditService
     {
         Task AddUserChange(AuditChangeDto auditChangeDto);
-        Task<Result<AuditChangeDto>> GetUserByTime(string userId, DateTime timestamp);
+        Task<Result<AuditUserInfoChangeDto>> GetUserByTime(string userId, DateTime timestamp);
     }
 
-    public class AuditService(
-        ILogger<AuditService> _logger,
-        IAuditRepository _auditRepository,
-        IMapper _mapper
-    ) : IAuditService
+    public class AuditService : IAuditService
     {
+        private readonly ILogger<AuditService> _logger;
+        private readonly IAuditRepository _auditRepository;
+        private readonly HttpClient _httpClient;
+        private readonly string _accountUserInfoUrl;
```

Comment:
Create typed Http client, for example using default Microsoft http clients or Refit library. Then u inject IHttpClientFactory and then CreateHttpClient("ClientNameHere"). This way you can register multiple clients and create them using factory, it's best practice

Reply:
And create your own wrapper for this HTTP client. This way you will be using it like a normal service in your code, and you will not even know that it's executing HTTP calls